stablehlo/dialect/VhloOps.cpp

```diff
@@ -296,5 +298,99 @@ void VhloDialect::printAttribute(Attribute attr, DialectAsmPrinter& os) const {
   assert(succeeded(result));
 }
 
+///////////////////////////
+// Op Constraint Versioning
+///////////////////////////
+// These could be migrated to ODS in VhloOps.td if we figured out a better way
+// to represent this sort of constraint in tablegen.
+
+namespace {
+bool checkIfOperandAndResultElementTypesMatch(TypeRange operandTypes,
+                                              TypeRange resultTypes) {
+  SmallVector<ShapedType> inputShapedTypes{
+      llvm::map_range(operandTypes, [](Type t) {
+        return convertTypeToBuiltinForPrint(t).cast<ShapedType>();
```

Comment:
@GleasonK Use of `convertTypeToBuiltinForPrint` might seem awkward as because of its name  "ForPrint". Let me know if we should rename it to `convertTypeToBuiltin`?

Reply:
Would prefer if we didn't convert to builtin at all? Can we make this a function of VHLO? I believe the initial changelist I shared had a method for extracting element type

Reply:
makes sense! updated. PTAL.